andela/valinor-ah-backend
# #162050416 Implement user account deactivate/delete feature

#### What does this PR do?
Implement the functionality to deactivate, re-activate and delete a user account

#### Description of Task to be completed?
- add status column to users table
- add endpoint `PATCH /api/v1/users/:userId/account/action` to modify account status by deactivating, activating or deleting
- ensure only a user can delete, deactivate, or activate his own account
- ensure email link is sent to user if they want to delete their account

#### How should this be manually tested?
- clone the repository, cd into it, checkout this branch `feature/162050416/deactivate-or-delete-account`
- set up your .env variables and local database
- run `npm install` to install required packages
- run `npm start` to start up the server
- use postman(or any api testing tool) to test the following routes route
- create a user using the route `POST http://localhost:3000/api/v1/users/signup` providing necessary request body.
- modify account status `PATCH http://localhost:3000/api/v1/users/:userId/account/:action` providing the token in the `Authorization` header as shown in screen shots below.
- the `action` param can either be `activate`, `deactivate` or `delete`.
- when you set action to delete, verify that you get an email in the address provided at signup
- click this link and verify that account is deleted by checking the db

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[162050416](https://www.pivotaltracker.com/story/show/162050416)

#### Screenshots (if appropriate)
deactivate my own account:
<img width="815" alt="screen shot 2018-11-20 at 10 55 17 pm" src="https://user-images.githubusercontent.com/31810086/48806176-d34ffc80-ed19-11e8-8ef9-8fa0f7a80c82.png">

activate my own account:
<img width="835" alt="screen shot 2018-11-20 at 10 55 30 pm" src="https://user-images.githubusercontent.com/31810086/48806181-d5b25680-ed19-11e8-850f-f4c51d8952ab.png">

modify another user's account:
<img width="841" alt="screen shot 2018-11-20 at 10 55 41 pm" src="https://user-images.githubusercontent.com/31810086/48806189-da770a80-ed19-11e8-83f7-507aec3d5bf4.png">

delete my own account (get email link):
<img width="902" alt="screen shot 2018-11-20 at 10 54 53 pm" src="https://user-images.githubusercontent.com/31810086/48806203-e236af00-ed19-11e8-9ce5-c5035801a40a.png">

Email received:
<img width="743" alt="screen shot 2018-11-20 at 10 56 03 pm" src="https://user-images.githubusercontent.com/31810086/48806195-dd71fb00-ed19-11e8-838b-5dcbf6ad0a1d.png">

Clicked email link:
<img width="742" alt="screen shot 2018-11-20 at 10 56 28 pm" src="https://user-images.githubusercontent.com/31810086/48806209-e5319f80-ed19-11e8-9a37-a460a2a07a14.png">

#### Questions:
N/A
